packages/nvidia-container-toolkit/nvidia-container-toolkit.spec

```diff
@@ -49,8 +49,8 @@ Conflicts: %{name}-ecs
 
 %build
 %cross_go_configure %{goimport}
-go build -buildmode=pie -ldflags="${GOLDFLAGS}" -o nvidia-container-runtime-hook ./cmd/nvidia-container-runtime-hook
-go build -buildmode=pie -ldflags="${GOLDFLAGS}" -o nvidia-ctk ./cmd/nvidia-ctk
+go build -ldflags="${GOLDFLAGS}" -o nvidia-container-runtime-hook ./cmd/nvidia-container-runtime-hook
```

Comment:
I'm not finding a great place to ask this, so this line is as good as any. Do we have a general idea on what *should* have FIPS and what shouldn't? I think we are finding them all in this PR but I'm unsure how we document what one needs to consider when the next Go package needs to be added to the repo. Is it just "always do these things for all Go packages?" If so, I think I'm fine with that but it might be nice to consider a way for us to avoid having so much repetitive logic for non-FIPS and FIPS builds and to update our guidance for packaging Go-based packages so we don't collectively forget what thought went into this PR later down the line.

Reply:
> I'm not finding a great place to ask this, so this line is as good as any. Do we have a general idea on what _should_ have FIPS and what shouldn't?

The [check-fips](https://github.com/bottlerocket-os/bottlerocket-sdk/blob/develop/macros/check-fips) script handles this pretty well. It becomes a build error to include a Go binary that imports `crypto/tls` without also building a FIPS-enabled version of that binary. At that point the packager has to address it. Some packages, like `oci-add-hooks`, are fine and don't need any special treatment.

I didn't want to gate the use of a new SDK on a PR like this, so the check is currently off by default. Once this merges I can flip that around in the next SDK release.

Reply:
The lede cites [check-fips](https://github.com/bottlerocket-os/bottlerocket-sdk/blob/develop/macros/check-fips) as the motivation. Is the criterion "if the binary uses crypto/tls, it needs a FIPS-equivalent using goboringcrypto"?

In the check-fips script, the test applied to the FIPS binary is:

```bash
      if ! uses_go_boring_crypto "${f}"; then
```

Is this sufficient, or should the script test:

```bash
      if uses_go_crypto_tls "${f}" then
        echo "${f} is built with crypto/tls" >&2
        (( miscompiled+=1 ))
      elif ! uses_go_boring_crypto "${f}"; then
        echo "${f} is not built with FIPS crypto (but ${b} is built with crypto/tls)" >&2
        (( miscompiled+=1 ))
      fi
```

I may be overcomplicating this, but I have encountered the "we linked library good-fred, but we also linked library fred-evil-twin, and the TLA customers DO NOT WANT" problem, so I worry.

Reply:
> I didn't want to gate the use of a new SDK on a PR like this, so the check is currently off by default. Once this merges I can flip that around in the next SDK release.

This is great, I think this is what I was hoping to see to make sure we don't have drift from the good state after this PR as we make changes to the Go packages.

Reply:
> I may be overcomplicating this, but I have encountered the "we linked library good-fred, but we also linked library fred-evil-twin, and the TLA customers DO NOT WANT" problem, so I worry.

The Go upstream uses (such as [boring.go](https://github.com/golang/go/blob/master/src/crypto/tls/boring.go) and [notboring.go](https://github.com/golang/go/blob/master/src/crypto/tls/notboring.go)) are gated behind conditionals like:
```
//go:build boringcrypto
//go:build !boringcrypto
```

That is pretty fundamental to how `GOEXPERIMENT=boringcrypto` is implemented so I am not very worried about including both - the build would fail because of multiple definitions for the same function, for example.